src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -213,13 +213,13 @@ else if (planKey.endsWith("-" + BuildPlanType.SOLUTION.getName())) {
      * @return the ResponseEntity with status 200 (OK) and the list of results in body
      */
     @GetMapping("exercises/{exerciseId}/results")
-    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @PreAuthorize("hasRole('TA')")
```

Comment:
Can you please check if this endpoint gets only called for this purpose (statistics)?
If yes: I suggest specifying its usage.
If no: Please check if this change makes sense for all other occurrences. If not, we should not use the same endpoint for both.

Reply:
From having a brief look at it it actually seems as if this would really only be used within `getResultsForExercise`, used in `getResults` in `result.service` which is only used in the `exercise-scores.component` which is used at many places. 

The call was not authorized for students for 5 days and I do not see further changes in https://github.com/ls1intum/Artemis/pull/4135/files. Changing the access rights was a misunderstanding there. - I do not think that it is beneficial to investigate this.
I would also not know a reason why a tutor should not be allowed to see the score of an exercise.

Reply:
In the backend it's less the score but the data that gets downloaded to the client. We might not want to share all data.

But since it's a revert then, okay